crates/alerter/src/main.rs

```diff
@@ -236,19 +243,30 @@ impl Alerter {
         tracing::debug!("found {} open orders", orders.len());
 
         std::mem::swap(&mut self.open_orders, &mut orders);
+        let mut closed_orders: Vec<Order> = orders.into_values().map(|(order, _)| order).collect();
         // Keep only orders that were open last update and are not open this update.
-        orders.retain(|order_uid, _| !self.open_orders.contains_key(order_uid));
-        for closed_order in orders.keys() {
-            tracing::debug!(order =% closed_order, "found closed order");
-            let order = self.orderbook_api.order(closed_order).await?;
-            if order.status == OrderStatus::Fulfilled {
+        closed_orders.retain(|order| !self.open_orders.contains_key(&order.uid));
+        // We're trying to find an order that has been filled. Try market orders first because they
+        // are more likely to be.
+        closed_orders.sort_unstable_by_key(|order| match order.class {
+            OrderClass::Market => 0u8,
+            OrderClass::Limit(_) => 1,
+            OrderClass::Liquidity => 2,
+        });
```

Comment:
Do we care about liquidity orders in the alerter at all? If not, they could also be filtered out in the `retain()` step.
As a side note for future reference: If we only care about bringing the market orders to the front `itertools::partition` could do that in linear time. But I see that it's currently not a dependency of the alerter and introducing that for this use case seems overkill.

Reply:
Just saw a few lines up outside the diff that liquidity orders already get filtered.